### DISCUSSION TITLE
`reply.send` Delay Issue in Fastify Requests

### DISCUSSION THREAD

**Discussion by ErikPham** [fastify/fastify#5617] [Q&A]
### Prerequisites

- [x] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.28.1

### Node.js version

18.20.4

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

3.19.3 (alpine)

### Description

We encountered an issue with Fastify where the `reply.send` function is delayed in sending the response. This is observed through the logs "Response html" and "Request completed."

Here is the example code:

```javascript
app.get('*', async function (req: FastifyRequest, reply: FastifyReply) {
    const reqId = req.id;
    reply.type('text/html; charset=utf-8');

    app.log.debug({ reqId }, 'Response html');
    const response = {
        statusCode: 200,
        html: '<!DOCTYPE html><html><head><title>Demo</title></head><body><div id="app"></div></body></html>'
    };
    return reply.status(response.statusCode).send(response.html);
});
```

We rely on the logs "Response html" and "Request completed" to identify the issue. The problem occurs when there is a significant delay between the invocation of reply.send and the actual response being sent, sometimes up to 30,000ms or even 40,000ms. We are unsure of the factors affecting the execution of reply.send.

**Comment by climba03003** [comment]
Note: The `completed` log is about the whole payload being completely sent, not start of it.

There are several reason that cause the high time difference between `.send` and `onResponse` hook.
1. too much blocking task on `onSend` hooks.
2. event loop is highly utilized
3. blocking task on other handler, stream write mostly process in `nextTick`. (need someone double confirm)
4. large payload
5. slow network
